examples/vision/swin_transformers.py

```diff
@@ -0,0 +1,557 @@
+"""
+Title: Image Classification with Swin Transformers
+Author: [Rishit Dagli](https://twitter.com/rishit_dagli)
+Date created: 2021/09/08
+Last modified: 2021/09/08
+Description: Image Classification using Swin Transformers, a general-purpose backbone for computer vision.
+"""
+"""
+This example implements [Swin Transformer: Hierarchical Vision Transformer using Shifted Windows](https://arxiv.org/abs/2103.14030)
+paper by Liu et al. for image classification, and demonstrates it on the 
+[CIFAR-100 dataset](https://www.cs.toronto.edu/~kriz/cifar.html).
+
+Swin Transformer (**S**hifted **Wi**ndow) capably serves as a general-purpose backbone 
+for computer vision. Swin Transformer is a hierarchical Transformer whose 
+representation is computed with shifted windows. The shifted windowing scheme 
+brings greater efficiency by limiting self-attention computation to 
+non-overlapping local windows while also allowing for cross-window connection. 
+This architecture also has the flexibility to model at various scales and has 
+linear computational complexity with respect to image size.
+
+This example requires TensorFlow 2.5 or higheras well as TensorFlow Addons, 
+which can be installed using the following commands:
+"""
+
+"""shell
+pip install -U tensorflow-addons
+"""
+
+"""
+## Setup
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+import tensorflow as tf
+import tensorflow_addons as tfa
+from tensorflow import keras
+from tensorflow.keras import layers
+from tensorflow.python.keras.layers.pooling import GlobalAveragePooling1D
```

Comment:
This is a private API, instead use `layers.GlobalAveragePooling1D`

Reply:
Got it, I always used to wonder what `.python.keras.*` is for, thanks